pkg/upstream/http.go

```diff
@@ -100,17 +101,37 @@ func newReverseProxy(target *url.URL, upstream options.Upstream, errorHandler Pr
 	if upstream.FlushInterval != nil {
 		proxy.FlushInterval = *upstream.FlushInterval
 	} else {
-		proxy.FlushInterval = 1 * time.Second
+		proxy.FlushInterval = options.DefaultFlushInterval
 	}
 
+	var finalTimeout time.Duration
+	if upstream.Timeout != nil {
+		finalTimeout = *upstream.Timeout
+	} else {
+		finalTimeout = options.DefaultTimeout
+	}
+
+	var finalKeepAlive time.Duration
+	if upstream.Timeout != nil {
+		finalKeepAlive = *upstream.KeepAlive
+	} else {
+		finalKeepAlive = options.DefaultKeepAlive
+	}
+
+	defaultTransportClone := http.DefaultTransport.(*http.Transport).Clone()
+	defaultTransportClone.DialContext = (&net.Dialer{
```

Comment:
I think we need to do similar logic for the `wsutil.ReverseProxy` we make for websockets here so this implementation has full coverage:

https://github.com/oauth2-proxy/oauth2-proxy/blob/aceb9e2762f41fe80f4364570572b2770440b4ca/pkg/upstream/http.go#L160

`wsutil.ReverseProxy.Dial` is public so we can set our own dialer pretty easily it looks like:

```
	// Dial specifies the dial function for dialing the proxied
	// server over tcp.
	// If Dial is nil, net.Dial is used.
	Dial func(network, addr string) (net.Conn, error)
```

Upon looking over this PR -- I have a strong suspicion my users who did log tailing user cases via websockets that ran into disconnections likely hit it due to these timeouts 😄